IOStream/AesOutputStream.m

```diff
@@ -50,9 +50,13 @@ - (NSUInteger)write:(const void *)bytes length:(NSUInteger)bytesLength {
     if (cs != kCCSuccess) {
         @throw [NSException exceptionWithName:@"EncryptError" reason:@"Failed to encrypt" userInfo:nil];
     }
+   
+    if( n > 0 ) {
+        // Write the encrypted data
+        [outputStream write:buffer length:n];
```

Comment:
The reason for this change is a bit subtle and took me a while to understand why it didn't work with KDBX4 files when it had worked fine with KDBX3.1 files.  In 3.1 files, when writing the XML to the file the sequence of stream operations are:

`XML -> GZIP -> HASHED BLOCKS -> AES ENCRYPT -> FILE`

however with KDBX4 files the order of stream ops is:

`XML -> GZIP -> AES ENCRYPT -> HMAC BLOCKS -> FILE`

The `GZIP` code was written with a check on the return value on the write:length: method such that it threw an Exception if the return value wasn't the number of bytes requested to be written.  For KDBX3.1 files, the `HashedOutputStream` class just returns the value sent into its write:length: method because it buffers the output until its memory buffer is full then it writes it all out in one large block.  The `AesOutputStream` class wasn't written that way and caused a problem when the stream order was as needed for KBDX4 files.

I added a check that throws an Exception if the downstream stream doesn't return the number of bytes requested.  However, the `HmacOutputStream` always returns the number of bytes requested just like `HashedOutputStream` because it just buffers the bytes in memory.